### PR TITLE
Added deployment tests

### LINKED ISSUE #227: deployment tests for autolabjs installation
The deployment tests are to be written ES6 compliant format using mocha and chai.

### PR DESCRIPTION
**Pull request by AnkshitJain**
## Description
This PR adds deployment tests for AutolabJS. 

## Motivation
 Closes #227 #225 #210 #209 #206 #202 #187 #19

 ## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checks
 - [x] Squash your commits down to one, or a few, discreet changesets before submitting a pull request
 - [x] Read the [contributing guide](https://github.com/AutolabJS/AutolabJS/blob/master/CONTRIBUTING.md)
 - [x] PR should be motivated, i.e. what does it fix, why, and if relevant how
 - [ ] If possible, include an example in the description, that could help all readers (including project members) to get a better picture of the change
 - [x] Avoid other runtime dependencies unless abslutely necessary
 - [x] Rebase your commit history so that each commit is meaningful and states the intention behind the change
 - [x] Ensure that your pull request follows the [style guidelines](https://github.com/AutolabJS/AutolabJS/wiki/Coding-Standards)
 - [x] At least one commit should mention `Fixes #<issue number>` _if relevant_

## Notify reviewers
@prasadtalasila

### CODE REVIEW

**Comment by prasadtalasila** [comment]
@AnkshitJain 
The deployment tests leave behind 6 new users (including lab_author, but excluding root) and 12 new projects (excluding root/test project). This is undesirable. It is better to clean up after the deployment tests.

File: tests/deployment_tests/container.bats [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+@test "Gitlab Container Check" {
+    result=$(grep -c "^gitlab/gitlab-ce:10.1.4-ce.0 gitlab Up.* (healthy)$" container_status.txt)
+    [ "$result" -eq 1 ]
+}
+
+@test "Restart Script Check" {
+    sudo bash -c "bash ./helper_scripts/container/restart_script_check.sh"
+    result=$(sudo crontab -l -u root | fgrep -c '* * * * * bash /root/autolab-restart.sh')
```
Comment (prasadtalasila) [inline tests/deployment_tests/container.bats]: It is better to match just `bash /root/autolab-restart.sh` part.

File: tests/deployment_tests/data/docker_locale/Buyer.java [hunk 2]
```diff
@@ -18,11 +19,11 @@ public void setCash(int cash)
     this.cash=cash;
   }
 
-  public void buy(Seller seller)
```
Comment (prasadtalasila) [inline tests/deployment_tests/data/docker_locale/Buyer.java]: Please keep this as seller. The change needs to be reflected in `if(x.sell()==1)` as well.

Comment (AnkshitJain) [reply tests/deployment_tests/data/docker_locale/Buyer.java]: This file is not renamed. It is directly taken from the docs/examples directory.

**Pull request merged**